File: app/src/main/java/com/example/neon/deardiary/data/source/DataSource.java
package com.example.neon.deardiary.data.source;

import com.example.neon.deardiary.data.Diary;

import java.util.ArrayList;
import java.util.Calendar;

public interface DataSource {
    interface LoadDiaryCallBack {
        void onDiaryLoaded(ArrayList<Diary> diaryList);

        void onDataNotAvailable();
    }

    interface GetDiaryCallBack {
        void onDiaryGot(Diary diary);

        void onDataNotAvailable();
    }

    interface UpdateDiaryCallback{
        void onDiaryUpdated();
        void onDiaryUpdateFailed();
    }

    void insertDiary(Diary diary);


    void updateDiary(Diary diary,UpdateDiaryCallback callback);

    void deleteAll();

    void queryByDay(Calendar calendar, GetDiaryCallBack callBack);

    void queryByMonth(Calendar calendar, LoadDiaryCallBack callBack);

    void queryAndAddDefault(Calendar calendar, LoadDiaryCallBack callBack);

    void queryByKeyword(String content, LoadDiaryCallBack callBack);

    int getValidDairyCount();
}

File: app/src/main/java/com/example/neon/deardiary/data/source/DataSource.kt
package com.example.neon.deardiary.data.source

import com.example.neon.deardiary.data.Diary

import java.util.ArrayList
import java.util.Calendar

interface DataSource {
    interface LoadDiaryCallBack {
        fun onDiaryLoaded(diaryList: ArrayList<Diary>)

        fun onDataNotAvailable()
    }

    interface GetDiaryCallBack {
        fun onDiaryGot(diary: Diary)

        fun onDataNotAvailable()
    }

    interface UpdateDiaryCallback {
        fun onDiaryUpdated()
        fun onDiaryUpdateFailed()
    }

    fun insertDiary(diary: Diary)

    fun updateDiary(diary: Diary, callback: UpdateDiaryCallback)

    fun deleteAll()

    fun queryByDay(calendar: Calendar, callBack: GetDiaryCallBack)

    fun queryByMonth(calendar: Calendar, callBack: LoadDiaryCallBack)

    fun queryAndAddDefault(calendar: Calendar, callBack: LoadDiaryCallBack)

    fun queryByKeyword(keyword: String, callBack: LoadDiaryCallBack)

    fun validDairyCount(): Int
}
